src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -136,10 +236,21 @@ export class SearchInterceptor {
         return throwError(new AbortError());
       }
 
-      const { combinedSignal, cleanup } = this.setupTimers(options);
+      const { timeoutSignal, combinedSignal, cleanup } = this.setupTimers(options);
       this.pendingCount$.next(this.pendingCount$.getValue() + 1);
 
-      return this.runSearch(request, combinedSignal, options?.strategy).pipe(
+      this.onRequestStart(request, options?.sessionId, timeoutSignal);
+
+      const res$ = this.runSearch(request, combinedSignal, options?.strategy);
```

Comment:
Do you need to `subscribe` here? Seems like you should just be able to do a `pipe`/`tap` instead, to still give the consumer the control of when to subscribe.